Revamp docs directory and add more docs (#21)
ScratchAddons/website-v2: content/docs/getting-started/installing.md

---
title: Installing
weight: 2
---

## From extension stores

> Scratch Addons is available in both Chrome Web Store and Add-ons for Firefox.

Reviewer: ```suggestion
Scratch Addons is available in both Chrome Web Store and addons.mozilla.org.
```
"addons.mozilla.org" is the official name of the extension distribution platform for Firefox.

- Chrome Web Store (for Google Chrome, Opera, Brave, Vivaldi, Microsoft Edge (>79), and other Chromium-based browsers)  
  https://chrome.google.com/webstore/detail/fbeffbjdlemaoicjdapfpikkikjoneco  

> - Add-ons for Firefox (for Mozilla Firefox)  

Reviewer: ```suggestion
- addons.mozilla.org (for Mozilla Firefox)  
```

  https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/  

## From GitHub

[The releases tab](https://github.com/ScratchAddons/ScratchAddons/releases) contains all builds of Scratch Addons, as well as the mirror of the store builds hosted on both servers.

## From source

### Cloning the repository

This is the recommended way to install for development purposes. This assumes you have Git installed.

To download the repository, simply clone `https://github.com/ScratchAddons/ScratchAddons.git`. Here's an example.

```sh
$ git clone https://github.com/ScratchAddons/ScratchAddons.git
```

> If you have GitHub, you can fork the repository in GitHub beforehand, and then clone the fork instead.

Reviewer: Drop this section. Users should always clone upstream repository.

```sh
$ git clone https://github.com/[your-name]/ScratchAddons.git
```

### Downloading the zipball

If you don't have Git installed, you can try this method instead.

> 1. Go to the [repository](https://github.com/ScratchAddons/ScratchAddons) and find the download code button

Reviewer: dot?

> ![Screenshot 2020-09-04 at 6 41 12 PM](https://user-images.githubusercontent.com/61319150/92291572-f17ef080-eede-11ea-85e8-fda961a56074.png)

Reviewer: Proper alt text?

Author: This, along with a heck ton of pics, probably could be solved later. For now it's fine.

2. Click it and select "Download ZIP"

![Screenshot 2020-09-04 at 6 43 57 PM](https://user-images.githubusercontent.com/61319150/92291570-f0e65a00-eede-11ea-80ee-f32e340327d9.png)

3. Extract the archive into a folder.

### Installing on Google Chrome

> 1. Go to the [Extension Management page](chrome://extensions) (`chrome://extensions`).

Reviewer: The page is not web-navigatable, and I don't think GFM allows that protocol. Unlinkify.

2. Click the toggle next to `Developer mode` to turn on the developer mode. This allows you to install the addon from a file.

![image](https://user-images.githubusercontent.com/61319150/92291715-8550bc80-eedf-11ea-9ec1-6628c5b3fbc2.png)

3. You should see `Load unpacked`. Clicking it will allow you to select a folder to upload.

![image](https://user-images.githubusercontent.com/61319150/92291792-d3fe5680-eedf-11ea-9d62-8e02b262f575.png)

> 4. Select the folder 

Reviewer: ```suggestion
4. Select the folder.
```

5. The extension should now be loaded. 

### Installing on Mozilla Firefox

<!-- TODO: Add pictures -->

> 1. Go to the [Debugging page](about://debugging#/runtime/this-firefox) and click "This Firefox" (`about://debugging#/runtime/this-firefox`).

Reviewer: The page is not web-navigatable, and I don't think GFM allows that protocol. Unlinkify.

> 2. Click `Load Temporary Add-on...` 

Reviewer: Either add dots on 2 and 3 or drop on 1 and 4
